pkg/ebpf/events_engine.go

```diff
@@ -0,0 +1,81 @@
+package ebpf
+
+import (
+	"context"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/rules/engine"
+	"github.com/aquasecurity/tracee/types/protocol"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// engineEvents stage in the pipeline allows rules detection to be executed in the pipeline
+func (t *Tracee) engineEvents(ctx context.Context, in <-chan *trace.Event) (<-chan *trace.Event, <-chan error) {
+	out := make(chan *trace.Event)
+	errc := make(chan error, 1)
+	done := make(chan bool)
+	engineInput := make(chan protocol.Event)
+
+	engineOutput := engine.StartPipeline(t.config.EngineConfig, engineInput, done)
+
+	// TODO: in the upcoming releases, the rule engine should be changed to receive trace.Event,
+	// and return a trace.Event, which should remove the necessity of converting trace.Event to protocol.Event,
+	// and converting detect.Finding into trace.Event
+
+	go func() {
+		defer close(out)
+		defer close(errc)
+		defer close(engineInput)
+
+		for {
+			select {
+			case event := <-in:
+				id := events.ID(event.EventID)
+
+				// if the even is marked as submit, we pass it to the engine
+				if t.events[id].submit {
+					err := t.parseArguments(event)
+					if err != nil {
+						t.handleError(err)
+						continue
+					}
+
+					// pass the event to the sink stage, if the event is also marked as emit
+					// it will be sent to print by the sink stage
+					out <- event
+
+					// send the event to the rule event
+					engineInput <- event.ToProtocol()
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	go func() {
+		for {
+			select {
+			case finding := <-engineOutput:
+				event, err := FindingToEvent(finding)
```

Comment:
Be aware that eventually you will have to call:
```
if !t.shouldProcessEvent(event) {
    continue
}
```
in here before sending event to "out" channel. This will happen because the user might have choosen the event (Which is a signature) but they might be filtering the output of such event through args, for example. For now you're using the original event args, but eventually it will be the "finding data", right ? Then you might "filter" out detections as well, just like other events.

Reply:
If the feature will be a "tech preview" I think it is fine do have an issue and mark it for the next release (considering we are out of time for this one).

Reply:
It won't be the default solution, for now, the only way to use it will be purposely choose it, also will make sure to note it as experimental on the release notes/docs.